src/components/wallet/AccountSelector.tsx

```diff
@@ -1,53 +1,54 @@
-import React, { useState, useEffect, useCallback } from 'react'
+import React, { useCallback } from 'react'
 
+import { Asset } from '@thorchain/asgardex-util'
 import { Menu, Dropdown } from 'antd'
 
+import { Size as CoinSize } from '../../components/uielements/assets/assetIcon/types'
 import Label from '../../components/uielements/label'
-import { shortSymbol } from '../../helpers/tokenHelpers'
-import { UserAssetType } from '../../types/wallet'
-import DynamicCoin from '../shared/icons/DynamicCoin'
+import AssetIcon from '../uielements/assets/assetIcon'
 import { StyledCard, AssetWrapper, AssetInfoWrapper, AssetTitle } from './AccountSelector.style'
 
 // Multi-use 'account selector' component will have a data context
 // Dummy data
-const UserAssets: UserAssetType[] = [
-  { _id: '1', free: 99, frozen: 11, locked: 21, symbol: 'BNB-JST', name: 'Binance', value: 0.99 },
-  { _id: '2', free: 1034, frozen: 38, locked: 101, symbol: 'RUNE-1E0', name: 'Rune', value: 0.25 }
-]
-
-type Props = { symbol?: string; onChange?: Function; size?: string }
-const AccountSelector: React.FC<Props> = ({ symbol, onChange, size }): JSX.Element => {
-  const [symb, setSymb] = useState<string>(symbol || UserAssets[0].symbol)
-  useEffect(() => {
-    if (typeof onChange === 'function') {
-      onChange(symb)
-    }
-  }, [symb, onChange])
+// const UserAssets: UserAssetType[] = [
+//   { _id: '1', free: 99, frozen: 11, locked: 21, asset: 'BNB-JST', name: 'Binance', value: 0.99 },
+//   { _id: '2', free: 1034, frozen: 38, locked: 101, asset: 'RUNE-1E0', name: 'Rune', value: 0.25 }
+// ]
+
+type Props = {
+  asset: Asset
+  assets: Asset[]
+  onChange?: (asset: Asset) => {}
+  size?: CoinSize
+}
+
+const AccountSelector: React.FC<Props> = (props: Props): JSX.Element => {
+  const { asset, assets, onChange = (_) => {}, size = 'normal' } = props
 
   const menu = useCallback(
     () => (
       <Menu>
-        {UserAssets.map((asset: UserAssetType, i: number) => (
-          <Menu.Item key={i} onClick={() => setSymb(asset.symbol)}>
+        {assets.map((asset, i: number) => (
+          <Menu.Item key={i} onClick={() => onChange(asset)}>
             <div style={{ display: 'flex' }}>
-              <div>{shortSymbol(asset.symbol)}</div>&nbsp;
+              <div>{asset?.symbol ?? 'unknown'}</div>&nbsp;
               <div>NAME..</div>&nbsp;
-              <div style={{ marginLeft: 'auto' }}>({asset.free})</div>
+              <div style={{ marginLeft: 'auto' }}>($ 8.50)</div>
```

Comment:
is it always 8.50 ? oO

Reply:
That's just a placeholder. `AccountSelector` still needs a proper logic.